src/scripts/main.js

```diff
@@ -1 +1,78 @@
 'use strict';
+
+const logo = document.querySelector('.logo');
+const div1 = document.createElement('div');
+const div2 = document.createElement('div');
+const div3 = document.createElement('div');
+
+div1.setAttribute('data-qa', 'notification');
+div2.setAttribute('data-qa', 'notification');
+div3.setAttribute('data-qa', 'notification');
+
+const successMessage = 'First promise was resolved';
+const errorMessage = 'First promise was rejected';
+const successMessage2 = 'Second promise was resolved';
+const successMessage3 = 'Third promise was resolved';
+
+const promise1 = new Promise((resolve, reject) => {
+  logo.addEventListener('click', () => {
+    resolve(successMessage);
+  });
+
+  setTimeout(() => {
+    reject(new Error(errorMessage));
+  }, 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  logo.addEventListener('click', () => {
+    resolve(successMessage2);
+  });
+
+  logo.addEventListener('contextmenu', (ev) => {
+    ev.preventDefault();
+    resolve(successMessage2);
+  });
+});
+
+const promise3 = new Promise((resolve) => {
+  logo.addEventListener('click', () => {
+    logo.addEventListener('contextmenu', (ev) => {
+      ev.preventDefault();
+      resolve(successMessage3);
+    });
+  });
+
+  logo.addEventListener('contextmenu', (ev) => {
+    logo.addEventListener('click', () => {
+      ev.preventDefault();
+      resolve(successMessage3);
+    });
+  });
```

Comment:
you can create 2 variables leftClicked and rightClicked for example, with false value by default and change theirs values when click happens. And check if values of these variables are true if yes resolve promise

Reply:
I couldn`t handle it without creating two eventListeners for left-right clicks and two right-left clicks, so I decided to use 'mousedown' with buttons check, cuz it looks more optimized =)